controller/instance_manager_controller.go

```diff
@@ -1610,6 +1630,22 @@ func (imc *InstanceManagerController) createInstanceManagerPodSpec(im *longhorn.
 	return podSpec, nil
 }
 
+func (imc *InstanceManagerController) cleanupInstanceManagerOrphanInstances(im *longhorn.InstanceManager) error {
+	orphans, err := imc.ds.ListOrphansByNodeRO(im.Spec.NodeID)
+	if err != nil {
+		return err
+	}
+	for _, orphan := range orphans {
+		switch orphan.Spec.Type {
+		case longhorn.OrphanTypeEngineInstance, longhorn.OrphanTypeReplicaInstance:
+			if err := imc.ds.DeleteOrphan(orphan.Name); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
```

Comment:
Use `multierr` to record the err and continue the next deletions.

Reply:
Refined. Please help review this part again, thanks.